### PR TITLE
Avoid invoking ruby if in a rebase

### PR DESCRIPTION
**Pull request by jhawthorn**
Invoking ruby and loading lolcommits can be very slow. For example, on my (fairly fast) machine:

```
$ time ruby -r lolcommits -e ''
ruby -r lolcommits -e ''  0.61s user 0.10s system 99% cpu 0.716 total
```

This isn't usually an issue for a single commit, but it makes lolcommits
impossible to use with a rebase-driven workflow.

This adds an if guard to the post-commit hook installer, which entirely
skips running ruby when in the middle of a rebase.

### CODE REVIEW

**Comment by matthutchinson** [comment]
This is a good idea  but it has already been implemented. The [die_if_rebasing](https://github.com/mroth/lolcommits/blob/master/lib/lolcommits/runner.rb#L113`) method in the runner has this covered.

Is this not working for you? Or are you concerned that just by launching a ruby script in the commit hook (even if it dies right away) -- it is still taking too long ?

I don't think the installer is the best place to put this for our gem, since a lot of lolcommit users will already have their hook setup, and unless they disable/enable on their repo - they won't get this change.

An alternative is to manually adjust the post-commit hook yourself (after enabling). 

**Or** you could change the hook to launch a bash script in ./bin that does the check and then conditionally launches Ruby. That would have to work cross-platform too tho, since we support Windows/Unix & OSX

Basically i'd like to keep the post-commit hook code a one-liner (ideally), and having this check in Ruby helps us manage cross-platform problems.

**Comment by jhawthorn** [comment]
> Is this not working for you? Or are you concerned that just by launching a ruby script in the commit hook (even if it dies right away) -- it is still taking too long ?

Yeah. Launching ruby is slow. Launching ruby and requiring lolcommits and all dependencies (which is needed to get to `die_if_rebasing`) is way, way too slow as my time benchmark shows above.

I'm currently editing the hooks manually to add this change, but I'd love to be able to share this fix as my coworkers (and presumably many other devs) need this change in order to use lolcommits. This does work on Linux and OS X, would it be acceptable to just skip it on windows?

**Comment by matthutchinson** [comment]
Ok, I understand your need for it.  

Can you adjust your PR so that it only adds the `if .. fi` block in the installer when we are not running in on the Windows platform?  `Lolcommits::Platform.platform_windows?` is available (and there is an if branch just before the lines you changed that you could probably use).  I'd merge that and get a new gem release ready.

We are planning to move the plugins out our our main gem and into external gems, but we haven't started that project just yet.  With more dependencies out of the way, hopefully the core gem will boot faster.

Another option would be to do this (and other) pre-checks in Ruby, but before requiring the main libraries and Gem dependencies.  

@mroth do you have any opinions on this one?

**Comment by mroth** [comment]
My initial feeling was to keep all the logic possible out of the stub and in the main program, but I think that ship has already sailed since we're parameterizing other things in the stub currently.  I'd prefer we be able to remove the Ruby version of this check then and not have two code paths executing that do the same thing, but unless we can make the wrapper work on Windows that will be problematic. Is there a reason we can't make the shell wrapper solution work on Windows as well?

I agree when we get the plugins out that boot time should become much faster, so we may want to TODO to revisit this then.  Recent versions of rubygems/bundler also had significant speed improvements for environments with lots of gems (see some of the talks @tenderlove has been giving lately, very interesting!).

**Comment by matthutchinson** [comment]
I think we could probably craft a Windows specific check for this file too, and provide both versions in the installed hook code rather - then we could remove the `die_if_rebasing` method.

I'll see if I can dust off my trusty Windows VM for this. @jhawthorn if you adjust your PR as I mentioned above, i'll merge and then try to throw in a Windows compatible check too.

**Comment by matthutchinson** [comment]
I've just (finally!) dusted off my Windows 7 VM.  First I tried to get [this](http://ss64.com/nt/if.html) CMD if statement to work in the post-commit, but no dice.

Then I found that (surprisingly) the bash `if [ ];` syntax works!  Whatever weird form the Windows git post-commit and `#!/bin/sh` combo shell takes, its all fine.

I will merge this now and then clean out the `die_if_rebasing` method in another commit, since it's no longer needed. Sorry for the much delayed response to this. I'm planning a new gem release (that will include this change) for the new year.
